tests/integration/controllers/suite_test.go

```diff
@@ -0,0 +1,160 @@
+package controllers
+
+import (
+	"context"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/config"
+	"github.com/harvester/harvester/pkg/controller/master/addon"
+	"github.com/harvester/harvester/pkg/server"
+	"github.com/harvester/harvester/tests/framework/cluster"
+	. "github.com/harvester/harvester/tests/framework/dsl"
+	"github.com/harvester/harvester/tests/integration/controllers/fake"
+	helmv1 "github.com/k3s-io/helm-controller/pkg/apis/helm.cattle.io/v1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher/lasso/pkg/controller"
+	"github.com/rancher/wrangler/pkg/generic"
+	batchv1 "k8s.io/api/batch/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	testSuiteStartErrChan chan error
+	testCtx               context.Context
+	testCtxCancel         context.CancelFunc
+	harvester             *server.HarvesterServer
+
+	kubeConfig       *restclient.Config
+	KubeClientConfig clientcmd.ClientConfig
+	testCluster      cluster.Cluster
+	options          config.Options
+	cfg              *rest.Config
+	scaled           *config.Scaled
+	testEnv          *envtest.Environment
+	scheme           = runtime.NewScheme()
+	crdList          = []string{"./manifest/helm-crd.yaml", "../../../deploy/charts/harvester-crd/templates/harvesterhci.io_addons.yaml"}
+)
+
+const (
+	harvesterStartTimeOut = 20
+)
+
+func TestAPI(t *testing.T) {
+	defer GinkgoRecover()
+
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "api suite")
+}
```

Comment:
Note: not required for the rc2, but seems we can merge this one with the API tests, additionally should we try to deploy the official [helm-controller](https://github.com/k3s-io/helm-controller/blob/master/manifests/deploy-cluster-scoped.yaml) to the kind cluster so we don't need to mimic the job controller and doing a real integration test to detect any potential issues from it, thanks.

Reply:
yeah i agree.. initially i wanted it to be separate to avoid tainting failures from the existing api testing.